dashboard/src/components/Catalog/Catalog.tsx

```diff
@@ -107,7 +107,7 @@ function Catalog(props: ICatalogProps) {
   useEffect(() => {
     const newFilters = {};
     Object.keys(propsFilter).forEach(filter => {
-      newFilters[filter] = propsFilter[filter]?.toString().split(",");
+      newFilters[filter] = propsFilter[filter]?.toString().split("|");
```

Comment:
I think we can leave the `,` as a separator since it's a common practice, we don't need to change that. The thing is that we don't support (or we don't want to support?) arrays for the text search. I would just ignore the `,` for the text search here:

```js
newFilters[filter] = filterNames.SEARCH === filter ? propsFilter[filter]?.toString() : propsFilter[filter]?.toString().split(",");
```

Reply:
Also, we need a test :)

Reply:
I don't think it is a good solution. Indeed, I did notice this problem when filtering operators; look at this comment: https://github.com/kubeapps/kubeapps/pull/2221#issuecomment-760943588

What do you think? If implementing your solution, we will still have the same issue in operators (and there are plenty of companies with "," in their names...)

Reply:
got it, then it's better to properly encode the values rather than just making the bug less probable

Reply:
(oh no, encoding again!! :roll_eyes: ) 
According to the RFC 3986?, the characters that may be subject to URL %-encoding are:
`!     #     $     %     &     '     (     )     *     +     ,     /     :     ;     =     ?     @     [     ]` 

But, they are considered sub-delimiters and there is no actual need to encode them even if encodeURIComponent does.
However, the `qs` parser doesn't support it yet (https://github.com/ljharb/qs/issues/237).

Indeed, `qs` is decoding all the parameters, so we don't have control over the values we received.
For instance, even if I push the filters properly like `Provider=something1,some%2Cthing2`, the component will always receive `something1,some,thing2`, so the split/join won't work.
So, I don't think encoding it's a straightforward solution, perhaps we should dig deeper into the `qs` module and re-implement the parsing function or directly substitute `qs` with another alternative.

I'd come to an agreement with the delimiter character to use and would leave this PR as is (perhaps adding the ignore you mentioned for the text search as well) since it is "workarounding" an existing bug that prevents us from releasing.

Anyway, I'm open to implementing other alternatives, of course :)

Reply:
> (oh no, encoding again!!  )

:laughing:

Reply:
okay, so I would still like to maintain the `,` as a separator because it's the standard but it's true that encoding won't help. Let's replace `,` in the filters with a highly improbable string (I suggest a double underscore `__`) and then undo the change when reading the filter. WDYT?

Reply:
Yes, I totally agree with this approach. Just a nitpick comment: if we use a non-to-be-encoded character like `_`, the URL will be like: ` ...?Provider=Lightbend__%20Inc` (the underscore is visible).
What if we use another symbol that should be encoded', for instance, the whitespace character in braille: `⠀` (U+2800)  (%E2%A0%80)
Anyway, I'm gonna start working with "__" and will update the PR asap.

Reply:
I don't mind what characters to use, use what you think is best